samples/visual_pane/components/visual_pane.js

```diff
@@ -0,0 +1,71 @@
+AFRAME.registerComponent("visual-pane", {
+  schema: {
+    jsonData: {
+      parse: JSON.parse,
+      stringify: JSON.stringify,
+    },
+  },
+
+  multiple: true,
+
+  init: function () {
+    const data = this.data.jsonData;
+    const el = this.el;
+
+    this.imageEl = document.createElement("a-image");
+    this.imageEl.setAttribute("id", "puzzle-piece-image");
+    this.imageEl.setAttribute("src", data.imageSrc);
+
+    const rawImageEl = document.querySelector(data.imageSrc);
```

Comment:
Ahhh, this may be problematic when integrating with backend. In the actual backend code, this is going to be a direct img URL (to our S3), rather than a local reference (#) to a-asset. Can you update the HTML's imgSrc to directly be `./pic2.jpg` and then still get this to work?

Reply:
Ah I see what you mean. To get the image dimensions I had to load it into an `<img>` tag first and then read from it. I'll see if there is a way to read directly without creating img tag in that case.

Reply:
I am not able to find a very good alternate way .. one way I saw was using javascript to create an `Image` object, and then doing the work in its `onLoad` function but there were comments saying it might not always work depending on browser and version..

What do you think about preloading this `img` tag under Aframe-assets as part of the server-side rendering?

Reply:
I would prefer to avoid doing special things for a component, if possible, so we can have a generic way for each component. Otherwise, we would have to loop through all objects every time. I can think of a couple options that we can explore:
1. Is the only thing you need, the image's width and height? Are these based on the original image upload? If they are, then we can just add them to jsonData and add the width/height of the image as part of the logic that saves the image originally. So, on the admin interface, admin would upload the image and when we are saving it as part of the scene, we automatically append the width and height. Would this work?
2. I believe we should be able to add an `<img>` dynamically using JS? Like:
```
var img = document.createElement("img");
img.src = "http://www.google.com/intl/en_com/images/logo_plain.png";
```
and then append it to the HTML page if needed:
```
var src = document.getElementById("header");
src.appendChild(img);
```
Since the `<img>` tag will exist on the page you should be able to get the width and height I believe.

Let me know if any of the above need more clarifications, or if both solutions above won't work, in which case I will be good with adding conditionals on serverside rendering 😛

Reply:
Yes, the width and height is all that is needed. This is just to figure out the ratio of the image. I like both suggestions.

I tried out `option 2`. It seems to work fine at the moment.. I guess we can switch to 1 if we encounter any issues